kura/test/org.eclipse.kura.container.orchestration.provider.test/src/test/java/org/eclipse/kura/container/orchestration/provider/EnforcementSecurityTest.java

```diff
@@ -0,0 +1,198 @@
+import org.eclipse.kura.container.orchestration.ContainerInstanceDescriptor;
+import org.eclipse.kura.container.orchestration.ContainerState;
+import org.eclipse.kura.container.orchestration.ImageConfiguration;
+import org.eclipse.kura.container.orchestration.PasswordRegistryCredentials;
+import org.eclipse.kura.container.orchestration.provider.impl.ContainerOrchestrationServiceImpl;
+import org.eclipse.kura.container.orchestration.provider.impl.enforcement.AllowlistEnforcementMonitor;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.ListImagesCmd;
+import com.github.dockerjava.api.command.StopContainerCmd;
+import com.github.dockerjava.api.model.Event;
+import com.github.dockerjava.api.model.Image;
+
+public class EnforcementSecurityTest {
+
+    private static final String IMAGE_NAME = "nginx";
+    private static final String CONTAINER_NAME = "frank";
+    private static final String CONTAINER_ID = "1d3dewf34r5";
+
+    private static final String REGISTRY_URL = "https://test";
+    private static final String REGISTRY_USERNAME = "test";
+    private static final String REGISTRY_PASSWORD = "test1";
+
+    private static final String EMPTY_ALLOWLIST_CONTENT = "";
+    private static final String FILLED_ALLOWLIST_CONTENT_NO_SPACE = "sha256:f9d633ff6640178c2d0525017174a688e2c1aef28f0a0130b26bd5554491f0da,sha256:c26ae7472d624ba1fafd296e73cecc4f93f853088e6a9c13c0d52f6ca5865107";
+    private static final String FILLED_ALLOWLIST_CONTENT_WITH_SPACES = " sha256:f9d633ff6640178c2d0525017174a688e2c1aef28f0a0130b26bd5554491f0da , sha256:c26ae7472d624ba1fafd296e73cecc4f93f853088e6a9c13c0d52f6ca5865107";
+
+    private static final String CORRECT_DIGEST = "ubuntu@sha256:c26ae7472d624ba1fafd296e73cecc4f93f853088e6a9c13c0d52f6ca5865107";
+    private static final String WRONG_DIGEST = "ubuntu@sha256:0000000000000000000000000000000000000000000000000000000000000000";
+
+    private AllowlistEnforcementMonitor allowlistEnforcementMonitor;
+    private ContainerOrchestrationServiceImpl mockedContainerOrchImpl;
+
+    ContainerConfiguration containerConfig;
+
+    private Map<String, Object> properties = new HashMap<>();
+
+    public EnforcementSecurityTest() {
+        this.properties.clear();
+    }
+
+    @Test
+    public void shouldAllowStartingWithCorrectAllowlistContent() throws KuraException, InterruptedException {
+
+        givenMockedContainerOrchestrationService();
+        givenMockedDockerClient(new String[] { CORRECT_DIGEST });
+        givenAllowlistEnforcement(FILLED_ALLOWLIST_CONTENT_NO_SPACE);
+
+        whenOnNext();
+
+        thenContainerDigestIsVerified();
+    }
+
+    @Test
+    public void shouldAllowStartingWithCorrectAllowlistContentWithSpaces() throws KuraException, InterruptedException {
+
+        givenMockedContainerOrchestrationService();
+        givenMockedDockerClient(new String[] { CORRECT_DIGEST });
```

Comment:
I think you're backing yourself into a corner here.

For the future I suggest you to have more flexible methods that renders the boundary conditions more explicit.

Example:

```suggestion
        givenMockedContainerOrchestrationService();
        givenMockedContainerOrchestrationServiceWith(CONTAINER_NAME, IMAGE_NAME, IMAGE_TAG, ContainerState.ACTIVE);
        givenMockedDockerClient(new String[] { CORRECT_DIGEST });
```

or something like that. Refer to the recent [ContainerInstance test refactoring](https://github.com/eclipse/kura/pull/5148) for ideas.